### ISSUE TITLE
ytmapi-rs: Error when song has no album

### ISSUE THREAD

**Issue by lebenoa** [nick42d/youtui#174]
Hello again. Sorry for opening another issue

When song has no album, the actual duration get "push" to album field and when we try to get duration field, we instead got an error

## Song which cause this

- `In The Dark (Extended Mix)` by `HALIENE & Andrew Rayel`

> I can provide JSON which cause this but it might be too long to paste in here

## Solution

Only solution I can think of is to check if album is duration or not

- ~~Band-aid fix: check if string contains `':'`~~ **As expected, some album contains `':'` but we can go for slower version instead** 

```rust
    let mut album = parse_flex_column_item::<String>(&mut mrlir, 1, delimiter_idx + 1)?;
    let is_duration = album.chars().all(|c| c.is_digit(10) || c == ':');
    let duration = if is_duration {
        let temp = album;
        album = String::new();
        temp
    } else {
        parse_flex_column_item(&mut mrlir, 1, delimiter_idx + 3)?
    };

```

- Actual fix: regex checking

**Comment by nick42d** [comment]
Hey there, thanks for the report.

It would indeed be useful to provide the JSON. Are you able to add it as an attachment?

I've also found it useful in the reports to get a screenshot sent through of what the youtube music page itself looks like. Is that something you'd be able to provide also? I think I can add an issue template to make it easier in future.

**Comment by lebenoa** [comment]
Yes, Gladly.

## JSON:
[ytmapi-rs.output.json](https://github.com/user-attachments/files/17525620/1.In.The.Dark.Extended.Mix.json)

## Image

> Song in question. Looks like one that doesn't pass QC from YouTube but we should be safe

![image](https://github.com/user-attachments/assets/25bb5e1f-9d92-4e5b-bf75-4299dc29df93)

**Comment by nick42d** [comment]
G'day, thanks again for the detailed report. I've resolved this under #174. Since the number of elements is discovered to be variable, I now parse the flex columns into a String and then split on the delimiter, meaning I can use more standard iterator methods to see if we've got the right field. Should be merged and released shortly.
